frontend/test/pytest/test_jax_integration.py

```diff
@@ -307,6 +307,27 @@ def cost_fn(x, y):
         assert jnp.allclose(result1, 0.0)
         assert jnp.allclose(result2, 0.0)
 
+    def test_multiD_calls(self, backend):
+        """Test a jax.grad in combination with qjit on non-1D input parameters."""
+
+        @qjit
+        @qml.qnode(qml.device(backend, wires=1))
+        def circuit(p1, p2):
+            qml.RY(p1[0, 1], wires=0)
+            qml.RZ(p2[1, 0], wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        def cost_fn(p1, p2):
+            m1 = circuit(p1, p2)
+            m2 = circuit(p1, p2)
+            return m1 + m2
+
+        p1 = jnp.array([[0.1, 0.3, 0.5], [0.1, 0.2, 0.8]])
+        p2 = jnp.array([[0.3, 0.5], [0.2, 0.8], [0.2, 0.8]])
+        result = jax.grad(cost_fn)(p1, p2)
+        reference = qjit(grad(cost_fn))(p1, p2)
```

Comment:
I think the reference should not involve Catalyst, similar to the other tests in this class.

Reply:
Done